src/hotspot/share/opto/parse2.cpp

```diff
@@ -1550,9 +1577,65 @@ void Parse::do_if(BoolTest::mask btest, Node* c) {
   } else {
     adjust_map_after_if(untaken_btest, c, untaken_prob, next_block);
   }
+
+  if (do_stress_trap) {
+    stress_trap(iff, counter, incr_store);
+  }
+}
+
+void Parse::stress_trap(IfNode* orig_iff, Node* counter, Node* incr_store) {
+  // Search for an unstable if trap
+  CallStaticJavaNode* trap = nullptr;
+  for (int i = 0; i <= 1; ++i) {
```

Comment:
Should we check that it is `IfNode` and it has 2 output edges? May be assert?

Reply:
I added an assert.